### PR TITLE
Remove more json! usage

### PR DESCRIPTION
**Pull request by GnomedDev**

### CODE REVIEW

File: src/builder/create_components.rs [hunk 1]
```diff
@@ -17,19 +16,20 @@ pub enum CreateActionRow<'a> {
...
-                Self::Buttons(x) => serde_json::to_value(x).map_err(S::Error::custom)?,
-                Self::SelectMenu(x) => serde_json::to_value(vec![x]).map_err(S::Error::custom)?,
-                Self::InputText(x) => serde_json::to_value(vec![x]).map_err(S::Error::custom)?,
-            }
-        })
-        .serialize(serializer)
+        use serde::ser::SerializeMap as _;
+
+        let mut map = serializer.serialize_map(Some(2))?;
+        map.serialize_entry("kind", &1_u8)?;
```
Comment (arqunis) [inline src/builder/create_components.rs]: Shouldn't this be `type` instead of `kind`?

Comment (GnomedDev) [reply src/builder/create_components.rs]: Indeed, fixed.

File: src/builder/create_components.rs [hunk 2]
```diff
@@ -17,19 +16,20 @@ pub enum CreateActionRow<'a> {
...
-        .serialize(serializer)
+        use serde::ser::SerializeMap as _;
+
+        let mut map = serializer.serialize_map(Some(2))?;
+        map.serialize_entry("type", &1_u8)?;
+
+        match self {
+            CreateActionRow::Buttons(buttons) => map.serialize_entry("components", buttons)?,
+            CreateActionRow::SelectMenu(select) => map.serialize_entry("components", select)?,
+            CreateActionRow::InputText(input) => map.serialize_entry("components", input)?,
```
Comment (arqunis) [inline src/builder/create_components.rs]: These two should serialise as an array.
```suggestion
            CreateActionRow::SelectMenu(select) => map.serialize_entry("components", [select])?,
            CreateActionRow::InputText(input) => map.serialize_entry("components", [input])?,
```

Comment (GnomedDev) [reply src/builder/create_components.rs]: Fixed

**Review by arqunis — APPROVED** [review]

**Pull request merged**